ReportUtilityBills/resources/util.rb

```diff
@@ -125,3 +590,164 @@ def self.calculate_monthly_prorate(header, month)
+  rates.each do |rate|
+    # rates to skip
+    next if rate['sector'] != 'Residential'
+    next if !rate['enddate'].nil?
+    next if keywords.any? { |x| rate['name'].downcase.include?(x) } && skip_keywords
+
+    # fixed charges
+    if ['$/day', '$/year'].include?(rate['fixedchargeunits'])
+      next
+    end
+
+    # min charges
+    if ['$/day'].include?(rate['minchargeunits'])
+      next
+    end
+
+    # ignore blank fields
+    rate.each do |k, v|
+      rate.delete(k) if v.nil?
+    end
+
+    # map schedules and structures
+    structures = {}
+    rate.each do |k, v|
+      if ['eiaid'].include?(k)
+        rate[k] = Integer(Float(v))
+      elsif k.include?('schedule')
+        # all of a sudden some fields have an "L" character (?)
+        rate[k] = eval(v.gsub('L', '')) # arrays
+      elsif k.include?('structure')
+        rate.delete(k)
+
+        k, period, tier = k.split('/')
+        period_idx = Integer(period.gsub('period', ''))
+        tier_idx = nil
+        tier_name = nil
+        ['max', 'unit', 'rate', 'adj', 'sell'].each do |k2|
+          if tier.include?(k2)
+            tier_idx = Integer(tier.gsub('tier', '').gsub(k2, ''))
+            tier_name = k2
+          end
+        end
+
+        # init
+        if !structures.keys.include?(k)
+          structures[k] = []
+        end
+        if structures[k].size == period_idx
+          structures[k] << []
+        end
+        if structures[k][period_idx].size == tier_idx
+          structures[k][period_idx] << {}
+        end
+
+        begin
+          v = Float(v)
+        rescue # string
+        end
+
+        structures[k][period_idx][tier_idx][tier_name] = v
+      else # not eiaid, schedule, or structure
+        begin
+          rate[k] = Float(v)
+        rescue # string
+        end
+      end
+    end
+
+    rate.update(structures)
+
+    # ignore rates with demand charges
+    next if !rate['demandweekdayschedule'].nil? || !rate['demandweekendschedule'].nil? || !rate['demandratestructure'].nil? || !rate['flatdemandstructure'].nil?
+
+    # ignore rates without minimum fields
+    next if rate['energyweekdayschedule'].nil? || rate['energyweekendschedule'].nil? || rate['energyratestructure'].nil?
+
+    # ignore rates without a "rate" key
+    next if rate['energyratestructure'].collect { |r| r.collect { |s| s.keys.include?('rate') } }.flatten.any? { |t| !t }
+
+    # ignore rates with a "sell" key
+    next if rate['energyratestructure'].collect { |r| r.collect { |s| s.keys } }.flatten.uniq.include?('sell')
+
+    # ignore rates with units other than "kWh"
+    # next if rate['energyratestructure'].collect { |r| r.collect { |s| s.keys.include?('unit') } }.flatten.any? { |t| !t }
+    # next if rate['energyratestructure'].collect { |r| r.collect { |s| s['unit'] == 'kWh' } }.flatten.any? { |t| !t }
+    rate['energyratestructure'].collect { |r| r.collect { |s| s['unit'] = 'kWh' } } # this field is suddenly gone, so assume kWh?
```

Comment:
We should find out the answer to this before merging. I noticed that there are ~1000 additional JSONs now, so it's possible that some of them were being filtered out by this previously (and should continue to be filtered out).